pages/map.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from "react";
+import { GoogleMap, Marker, useJsApiLoader } from "@react-google-maps/api";
+import { zoneServiceUi } from "../ui/ZoneServices"
+import { InterestZone, Status } from "../models/zone.model";
+import { withPageAuthRequired } from '@auth0/nextjs-auth0';
+
+const containerStyle = {
+  width: "100vw",
+  height: "100vh",
+};
+
+const center = {
+  lat: 46.7554536,
+  lng: 23.5671444,
+};
+
+type GoogleLatLng = google.maps.LatLng;
+
+const yellowPin =  "/icon/in-progress.png";  
+const greenPin =  "/icon/done.png"; 
+const redPin =  "/icon/to-do.png";
+
+const libraries:("places")[] = ["places"];
+function GoogleMaps() {
+  const { isLoaded } = useJsApiLoader({
+    id: "google-map-script",
+    googleMapsApiKey: process.env.NEXT_PUBLIC_APP_GOOGLE_MAPS_API_KEY as any,
+    libraries,
+  });
+
+  const [map, setMap] = useState<GoogleMap>();
+  const [interestZones, setInterestZones] = useState<InterestZone[]>([]);
+
+  const getColorMarkerByStatus = (status:Status): string => {
+    let selectedPin = '';
+    switch (status) {
+      case Status.TODO:
+        selectedPin = redPin;
+        break;
+      case Status.INPROGRESS: 
+        selectedPin = yellowPin;
+        break;
+      case Status.DONE:
+        selectedPin = greenPin;
+        break;
+      default: break;
+    }
+    return selectedPin;
+  }
+
+  const onLoad = (map: any) =>  {
+    map.panTo(new google.maps.LatLng( 46.7554537,23.5671444))
+    setMap(map);
+  };
+
+  const loadAllPoints = async () => {   
+    return (await zoneServiceUi.findAll())
+   }
```

Comment:
Please try to avoid defining methods inside components, excepting for react hooks and callbacks. In this case, we can add the service call directly in the `useEffect` hook.